compiler-core/src/dependency.rs

```diff
@@ -57,6 +64,25 @@ where
     Ok(packages)
 }
 
+// If the string would parse to an exact version then return the version
+fn parse_exact_version(ver: &str) -> Option<Version> {
```

Comment:
This feels bad/unnecessary but it doesn't seem like there is a good way to get the version out of the pubgrub parse directly